openff/toolkit/topology/topology.py

```diff
@@ -94,16 +94,13 @@ class ValenceDict(_TransformedDict):
     @staticmethod
     def key_transform(key):
         """Reverse tuple if first element is larger than last element."""
-        # Ensure key is a tuple.
         key = tuple(key)
-        assert len(key) > 0 and len(key) < 5, "Valence keys must be at most 4 atoms"
-        # Reverse the key if the first element is bigger than the last.
```

Comment:
(not blocking) This PR opts to delete a lot of comments, a few of which are indeed incorrect, but many of which are correct-but-only-marginally-useful. In the future I'd kinda appreciate having fewer marginally-useful comments removed.

Reply:
I was always taught not to have low-level comments that are immediately obvious from the nearby code, like `Ensure key is a tuple` -- both because they introduce unneeded clutter and because they easily go out of date when the code is changed. I thought about leaving some of the comments in the "im/possible" section but then it seemed that most of them were covered in the docstring or associated error messages.